test/canary/canary_test.go

```diff
@@ -32,46 +30,83 @@ func init() {
 	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
 }
 
-// This copies the canary config, starts the agent, and confirm version is correct.
-// Sanity test runs earlier in the terraform code execution
+// TestCanary verifies customers can download the agent, install it, and it "works".
+// Reports metrics on the number of download failures, install failures, and start failures.
 func TestCanary(t *testing.T) {
```

Comment:
Why we would need to have the `downloaded_cwa.rpm` here if we can setup all the related things there and we only [need to provide the RPM](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/terraform/ec2/linux/main.tf#L66-L76)

Reply:
We want to repeatedly download from S3. Not just repeatedly install the already provided RPM.